src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7930,22 +7932,41 @@ public static LocalState UnreachableState
             {
                 get
                 {
-                    BitVector state = BitVector.Create(1);
+                    BitVector state = BitVector.Create(2);
                     state[0] = false;
                     return new LocalState(state);
                 }
             }
 
-            public int Capacity => _state.Capacity;
+            public int Capacity => _state.Capacity / 2;
 
-            public void EnsureCapacity(int capacity) => _state.EnsureCapacity(capacity);
+            public void EnsureCapacity(int capacity) => _state.EnsureCapacity(capacity * 2);
 
             public NullableFlowState this[int slot]
             {
-                get => (slot < Capacity && this.Reachable && _state[slot]) ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
-
-                // No states should be modified in unreachable code, as there is only one unreachable state.
-                set => _ = !this.Reachable || (_state[slot] = (value == NullableFlowState.MaybeNull));
+                get
+                {
+                    if (slot < Capacity && this.Reachable)
+                    {
+                        slot *= 2;
+                        return (_state[slot + 1], _state[slot]) switch
```

Comment:
>_state[slot + 1], _state[slot] [](start = 32, length = 30)

Do we ever use index 1?  You might consider using `slot-1` and `slot`.

Reply:
We could but that would complicate the `* 2` and `/ 2` elsewhere in the class.

---
In reply to: [347694067](https://github.com/dotnet/roslyn/pull/39778#discussion_r347694067) [](ancestors = 347694067)